File: core/src/main/java/com/infoa2/core/widget/impl/ZipCodeValidatorInputType.java
package com.infoa2.core.widget.impl;

import android.content.Context;
import com.infoa2.core.R;
import com.infoa2.core.widget.EditText;
import com.infoa2.core.widget.interfaces.ValidatorInputType;

/**
 * Created by caiomansho on 05/04/2017.
 */

public class ZipCodeValidatorInputType implements ValidatorInputType {

    private EditText editText;
    private Context context;

    public ZipCodeValidatorInputType(Context context, EditText editText) {
        this.editText = editText;
        this.context = context;

    }

    @Override
    public Boolean isValid() {
        if (!editText.getText().toString().isEmpty() && !editText.isZipCode()) {
            editText.setErrorEnabled(true);
            if (editText.getCustomErrorMessageId() != 0) {
                editText.setError(context.getString(editText.getCustomErrorMessageId()));

            } else {
                editText.setError(context.getString(R.string.sou_widgets_invalid_zip_code));

            }
            return Boolean.FALSE;

        } else if(editText.getText().toString().isEmpty()) {
            editText.setError(null);
            editText.setErrorEnabled(false);
            return Boolean.FALSE;

        } else {
            editText.setError(null);
            editText.setErrorEnabled(false);
            return Boolean.TRUE;

        }
    }

}

File: core/src/main/java/com/infoa2/core/widget/impl/ZipCodeValidatorInputType.kt
package com.infoa2.core.widget.impl

import android.content.Context
import com.infoa2.core.R
import com.infoa2.core.widget.EditText
import com.infoa2.core.widget.interfaces.ValidatorInputType

/**
 * Created by caiomansho on 05/04/2017.
 */

class ZipCodeValidatorInputType(private val context: Context, private val editText: EditText) : ValidatorInputType {

    override val isValid: Boolean
        get() {
            if (!editText.text!!.toString().isEmpty() && !editText.isZipCode) {
                editText.isErrorEnabled = true
                if (editText.customErrorMessageId != 0) {
                    editText.error = context.getString(editText.customErrorMessageId)

                } else {
                    editText.error = context.getString(R.string.sou_widgets_invalid_zip_code)

                }
                return java.lang.Boolean.FALSE

            } else if (editText.text!!.toString().isEmpty()) {
                editText.error = null
                editText.isErrorEnabled = false
                return java.lang.Boolean.FALSE

            } else {
                editText.error = null
                editText.isErrorEnabled = false
                return java.lang.Boolean.TRUE

            }
        }

}
